src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -813,7 +813,14 @@ private void EmitSideEffects(BoundSequence sequence)
             {
                 foreach (var se in sideEffects)
                 {
-                    EmitExpression(se, false);
+                    if (se is BoundExpression)
```

Comment:
>is BoundExpression [](start = 27, length = 18)

Consider using `as` or `is BoundExpression e` so the cast is not needed below.

Reply:
Will do, though the generated IL would be the same. The runtime compiler knows to do the type conversion once.